core/src/main/java/io/questdb/griffin/engine/table/parquet/PartitionDescriptor.java

```diff
@@ -24,68 +24,29 @@
 
 package io.questdb.griffin.engine.table.parquet;
 
-import io.questdb.cairo.SymbolMapWriter;
 import io.questdb.std.*;
 import io.questdb.std.str.DirectUtf8Sink;
 import io.questdb.std.str.Utf8Sequence;
 
+// This class manages memory for Parquet partition data.
+// It handles memory with a different lifetime than the PartitionDescriptor.
 public class PartitionDescriptor implements QuietCloseable {
-    private DirectLongList columnAddrs = new DirectLongList(16, MemoryTag.NATIVE_DEFAULT);
-    private DirectIntList columnIds = new DirectIntList(16, MemoryTag.NATIVE_DEFAULT);
-    private DirectIntList columnNameLengths = new DirectIntList(16, MemoryTag.NATIVE_DEFAULT);
-    private DirectUtf8Sink columnNames = new DirectUtf8Sink(32);
-    private DirectLongList columnSecondaryAddrs = new DirectLongList(16, MemoryTag.NATIVE_DEFAULT);
-    private DirectLongList columnSecondarySizes = new DirectLongList(16, MemoryTag.NATIVE_DEFAULT);
-    private DirectLongList columnSizes = new DirectLongList(16, MemoryTag.NATIVE_DEFAULT);
-    private DirectLongList columnTops = new DirectLongList(16, MemoryTag.NATIVE_DEFAULT);
-    private DirectIntList columnTypes = new DirectIntList(16, MemoryTag.NATIVE_DEFAULT);
-    private boolean memoryOwner;
-    private long partitionRowCount;
-    private DirectLongList symbolOffsetsAddrs = new DirectLongList(16, MemoryTag.NATIVE_DEFAULT);
-    private DirectLongList symbolOffsetsSizes = new DirectLongList(16, MemoryTag.NATIVE_DEFAULT);
-    private DirectUtf8Sink tableName = new DirectUtf8Sink(16);
-    private int timestampIndex = -1;
-
-    public void addColumn(
-            final CharSequence columnName,
-            int columnType,
-            int columnId,
-            long columnTop,
-            long columnAddr,
-            long columnSize,
-            long columnSecondaryAddr,
-            long columnSecondarySize,
-            long symbolOffsetsAddr,
-            long symbolOffsetsSize
-    ) {
-        final int startSize = columnNames.size();
-        columnNames.put(columnName);
-        columnNameLengths.add(columnNames.size() - startSize);
-        columnIds.add(columnId);
-        columnTops.add(columnTop);
-        columnTypes.add(columnType);
-        columnAddrs.add(columnAddr);
-        columnSizes.add(columnSize);
-        columnSecondaryAddrs.add(columnSecondaryAddr);
-        columnSecondarySizes.add(columnSecondarySize);
-        symbolOffsetsAddrs.add(symbolOffsetsAddr);
-        symbolOffsetsSizes.add(symbolOffsetsSize);
-    }
+    protected DirectLongList columnAddrs = new DirectLongList(16, MemoryTag.NATIVE_DEFAULT);
```

Comment:
nit: quite a lot of Java objects, a single DirectLongList would be enough where all these addresses are stored in a flat array